packages/block-editor/src/components/link-control/use-remote-url-data.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import makeCancelable from './make-cancelable';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useState, useEffect, useRef, useCallback } from '@wordpress/element';
+
+function useRemoteUrlData( url ) {
+	const cancelableFetch = useRef();
+
+	const [ richData, setRichData ] = useState( null );
+	const [ isFetching, setIsFetching ] = useState( false );
+
+	const { fetchRemoteUrlData } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return {
+			fetchRemoteUrlData: getSettings().__experimentalFetchRemoteUrlData,
+		};
+	}, [] );
+
+	const cancelPendingFetch = useCallback( () => {
+		if ( cancelableFetch.current ) {
+			cancelableFetch.current.cancel();
+		}
+	}, [ cancelableFetch ] );
+
+	/**
+	 * Cancel any pending requests that were made for
+	 * stale URL values. If the component does not unmount
+	 * we must handle cancelling the current request if
+	 * the URL changes otherwise we will see stale data.
+	 */
+	useEffect( () => {
+		cancelPendingFetch();
+	}, [ url ] );
+
+	/**
+	 * Cancel any pending requests on "unmount"
+	 */
+	useEffect( () => {
+		return () => {
+			cancelPendingFetch();
+		};
+	}, [] );
```

Comment:
I don't think any of these are necessary, we can just return a cleanup function in the effect below to call the cancel function instead.

Reply:
Weirdly the `<LinkPreview>` does not unmount when clicking between links. So if you click on a link and show the preview and then click immediately on another link, the popup doesn't unmount but rather simply renders down the tree.

This means that we need to actively cancel anything that's pending when the URL changes otherwise we will see stale data.

Reply:
An easier and probably more straightforward way is to force remount the component whenever `url` changed. That way we can reset all the states and effects to prevent weird things from happening.

Reply:
> An easier and probably more straightforward way is to force remount the component whenever url changed via the key attribute

Done 👍